app/src/main/java/com/mashup/mobalmobal/ui/profile/presenter/ProfileViewModel.kt

```diff
@@ -3,52 +3,143 @@ package com.mashup.mobalmobal.ui.profile.presenter
+    }
+
+    private val _itemSubject: BehaviorSubject<List<ProfileItem>> = BehaviorSubject.create()
+    val itemSubject: Observable<List<ProfileItem>> = _itemSubject.distinctUntilChanged()
+
+    init {
+        _userDtoSubject.combineLatest(_donationsSubject)
+            .distinctUntilChanged()
+            .map { listOf(it.first.toProfileItem()) + it.second }
+            .subscribeWithErrorLogger {
+                _itemSubject.onNext(it)
             }
             .addToDisposables()
     }
 
+    private fun requestDonations() =
+        requestMyCreateDonations()
+            .zipWith(requestMyDonated())
+            .zipWith(requestMyClosed())
+            .map {
+                val createDonations =
+                    it.first.first.toProfileItems(R.string.profile_header_donation_request)
+                val donatedDoantions =
+                    it.first.second.toProfileItems(R.string.profile_header_donated)
+                val closedDonations =
+                    it.second.toProfileItems(R.string.profile_header_donation_closed)
+
+                mutableListOf<ProfileItem>().apply {
+                    addAll(
+                        Triple(
+                            createDonations.getDonationCount(),
+                            donatedDoantions.getDonationCount(),
+                            closedDonations.getDonationCount()
+                        ).toSummaryProfileItem()
+                    )
+
+                    addAll(createDonations + donatedDoantions + closedDonations)
+                }
+            }
+
+    private fun requestMyCreateDonations() =
+        profileRepository.getMyPosts(STATUS_DONATION_BEFORE)
+            .zipWith(profileRepository.getMyPosts(STATUS_DONATION_INPROGRESS))
+            .map { pair ->
+                val beforeDonation = pair.first.data.posts
+                val inProgressDonation = pair.second.data.posts
+                val list = mutableListOf<PostDto>().also {
+                    if (!beforeDonation.isNullOrEmpty()) it.addAll(beforeDonation)
+                    if (!inProgressDonation.isNullOrEmpty()) it.addAll(inProgressDonation)
+                }
+                MyPostDto(list)
+            }
+
+    private fun requestMyDonated() =
+        profileRepository.getMyDonations()
+            .map {
+                if (it.data.donates.isNullOrEmpty()) {
+                    MyDonateDto(emptyList())
+                } else {
+                    it.data
+                }
+            }
+
+    private fun requestMyClosed() =
+        profileRepository.getMyPosts(STATUS_DONATION_EXPIRED)
+            .map {
+                if (it.data.posts.isNullOrEmpty()) {
+                    MyPostDto(emptyList())
+                } else {
+                    it.data
+                }
+            }
+
+    private fun List<ProfileItem>.getDonationCount() =
+        if (isNullOrEmpty()) {
+            0
+        } else {
+            lastIndex
+        }
```

Comment:
이 리스트 안에 Donation 아닌 item도 포함되어 있어서 이렇게 작성하신 건가요? 그렇다면 이것보단 
```suggestion
    private fun List<ProfileItem>.getDonationCount(): Int = filterIsInstance(ProfileItem.Donation::Class.java).size
```
이렇게 해서 확실히 체크하는게 좋습니다.
에러를 야기시킬 수 있습니다.

Reply:
오홍 좋습니다~